core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java

```diff
@@ -225,25 +225,40 @@ public void close() {
       LOG.info("Closing commit service for {}", table);
       committerService.shutdown();
 
+      boolean timeout = false;
+      int waitTime;
       try {
         // All rewrites have completed and all new files have been created, we are now waiting for
         // the commit
-        // pool to finish doing it's commits to Iceberg State. In the case of partial progress this
+        // pool to finish doing its commits to Iceberg State. In the case of partial progress this
         // should
         // have been occurring simultaneously with rewrites, if not there should be only a single
         // commit operation.
-        // In either case this should take much less than 10 minutes to actually complete.
-        if (!committerService.awaitTermination(10, TimeUnit.MINUTES)) {
+        // We will wait 10 minutes plus 5 more minutes for each commit left to perform due to the
+        // time required for writing manifests
+        waitTime = 10 + (completedRewrites.size() / rewritesPerCommit) * 5;
```

Comment:
[doubt] is the 5 min per commit, coming from present defaults i.e commit.retry.num-retries = 4, commit.retry.max-wait-ms = 1 min ?

Reply:
Can we change it to a longer duration, like 15 mins per commit? In that way, the waiting time increases proportionally instead of a fixed 120-mins. The question we are trying to answer is that, is 120 mins long enough?

Reply:
120 mins should be long enough, this is really just a temporary workaround while we get parallel commits implemented. I figured the less complicated the better

Reply:
That should be OK then.